### PR TITLE
improve code quality in fixedloadmodelspec and other tests

### LINKED ISSUE #919: Improve code quality in `FixedLoadModelSpec` and other tests
Things we want to avoid:
- using for/while loops (use `map`, `foreach`, `foldLeft` etc.)
- using `var` (usually can be prevented by folding)

Furthermore in `SystemComponentSpec`:
- using for loops for test cases, check `ProportionalFlexStratSpec` for a different way of implementing this

`WecModelSpec`:
- unnecessary usage of `new` with case class `WecRelevantData`
- test cases could be adapted as well, including last test that currently uses `temperatures.zip(expectedPowers)`

`ThermalHouseSpec`:
- test cases and foreach

More information on table-driven property checks [here](https://www.scalatest.org/user_guide/table_driven_property_checks). The advantage of using the `forAll` method is that in case of failure, information about the used parameters is logged as well.

Please let me know if there's any questions

### PR DESCRIPTION
**Pull request by pierrepetersmeier**
improve code quality in fixedloadmodelspec and other tests 

resolves #919

### CODE REVIEW

**Comment by pierrepetersmeier** [comment]
@sebastian-peter I hope everything fits now. Should all 'foreach' be converted to 'forall'?

File: src/test/scala/edu/ie3/simona/model/grid/SystemComponentSpec.scala [hunk 1]
```diff
@@ -103,12 +103,10 @@ class SystemComponentSpec extends UnitSpec with DefaultTestData {
         ),
       )
 
-      for ((operationStart, operationEnd, expected) <- testCases) {
+      testCases.foreach { case (operationStart, operationEnd, expected) =>
```
Comment (sebastian-peter) [inline src/test/scala/edu/ie3/simona/model/grid/SystemComponentSpec.scala:106]: Please use `forAll(testCases)` as well :)

File: src/test/scala/edu/ie3/simona/model/participant/WecModelSpec.scala
Comment (sebastian-peter) [inline src/test/scala/edu/ie3/simona/model/participant/WecModelSpec.scala:1]: Could you try to introduce `forAll(testCases)` here as well? So for "calculate active power output depending on velocity", there would be twelve test cases and for "calculate active power output depending on temperature", there would be three. Don't know if it's understandable what I mean, please just let me know if not.

File: src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala
Comment (sebastian-peter) [inline src/test/scala/edu/ie3/simona/model/thermal/ThermalHouseSpec.scala:1]: Yes, that's what I meant. Thanks!

File: src/test/scala/edu/ie3/simona/model/participant/load/FixedLoadModelSpec.scala [hunk 2]
```diff
@@ -116,8 +115,9 @@ class FixedLoadModelSpec
       forAll(testData) { (reference, expectedPower: Power) =>
         val relevantData = FixedLoadModel.FixedLoadRelevantData
 
-        var scale = 0.0
-        while (scale <= 2) {
+        val scales = (0 to 20).map(_ * 0.1)
```
Comment (sebastian-peter) [inline src/test/scala/edu/ie3/simona/model/participant/load/FixedLoadModelSpec.scala:118]: Could also probably be (0.0 to 2.0 by 0.1)

**Review by sebastian-peter — CHANGES_REQUESTED** [review]

File: src/test/scala/edu/ie3/simona/model/grid/SystemComponentSpec.scala [hunk 3]
```diff
@@ -103,12 +103,10 @@ class SystemComponentSpec extends UnitSpec with DefaultTestData {
         ),
       )
 
-      for ((operationStart, operationEnd, expected) <- testCases) {
+      testCases.forall { case (operationStart, operationEnd, expected) =>
```
Comment (sebastian-peter) [inline src/test/scala/edu/ie3/simona/model/grid/SystemComponentSpec.scala:106]: Please also use test cases with `forAll(testCases)` here. For this to work, the test cases have to be defined as a `Table`.

File: src/test/scala/edu/ie3/simona/model/grid/SystemComponentSpec.scala [hunk 4]
```diff
@@ -115,8 +115,8 @@ class SystemComponentSpec extends UnitSpec with DefaultTestData {
             simulationEnd,
             operationTime,
           )
-        interval should be(expected)
-      }
+        interval == expected
+      } shouldBe true
```
Comment (sebastian-peter) [inline src/test/scala/edu/ie3/simona/model/grid/SystemComponentSpec.scala:119]: ... then here you'd still use `should be`

**Review by sebastian-peter — CHANGES_REQUESTED** [review]
Thanks for considering my requests. I only have one more pedantic wish.

**Pull request merged**